### PR TITLE
Add support for extracting referenced enum types

### PR DESCRIPTION
**Pull request by papandreou**
The DB schema I'm working with is using a lot of enum types. When I ran `pg-to-dbml` on it, all of those came out as `USER-DEFINED` in the dbml output.

This adds support for extracting the enum values from all referenced enums and rendering them as DBML. For example:

```sql
CREATE SCHEMA myschema;
SET search_path TO 'myschema';
CREATE TYPE my_enum AS ENUM('foo', 'bar', 'quux');
CREATE TABLE my_things (thing_id BIGINT, status my_enum);
```

This comes out as:

```
Enum "my_enum" {
        "bar"
        "foo"
        "quux"
}

Table "my_things" {
        "thing_id" bigint   
        "status" my_enum   
 } 
```

I tested it with dbdocs, which does pick it up nicely: https://dbdocs.io/andreas.lind/myschema?table=my_things&schema=public&view=table_structure

Screenshot 2022-03-15 at 08 39 52

Note: Postgres does not support comments on enum _values_ (even though DBML does). Postgres and DBML _do_ support comments on types. However, I couldn't get dbinfo.io to accept the syntax, so I've left that out for now 😇

### CODE REVIEW

**Comment by camoore1979** [comment]
@papandreou Thank you for all of your recent PRs!! We are super busy right now (as everyone is, I'm sure) but we will try to get these resolved ASAP. These are issues our team has encountered as well, so your contributions are very welcome.

There is some issue with our lint configuration, soon I hope to resolve that as well.

**Comment by papandreou** [comment]
@camoore1979, nice! Looking forward to it, although I'm not in a hurry. Let me know if there's anything I can do.

File: src/funcs/getColumnDefinition.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const getColumnType = require('./getColumnType');
 const getColumnSettings = require('./getColumnSettings');
 
-module.exports = function getColumnDefinition(col) {
+module.exports = function getColumnDefinition(col, primaryKeys) {
```
Comment (papandreou) [inline src/funcs/getColumnDefinition.js:4]: This is just a drive-by at this point. I thought I'd need to pass some info about the available `enum` types in here, and then noticed that `primaryKeys` [was being passed in](https://github.com/polarislabs/pg-to-dbml/blob/450ea07b86526654c69e95b239e3075310d9ef45/src/funcs/transformTableStructureToDBML.js#L10) as the 2nd parameter already.

**Comment by papandreou** [comment]
I guess we could also consider rendering the enum names "fully qualified" as `schema_name.enum_name`. That's at least supported by the DBML spec: https://www.dbml.org/docs/#enum-definition

It's probably a very rare occurrence that you'll have different enum types with the same name residing in different schemas, though 🤔

**Comment by papandreou** [comment]
@camoore1979, still busy? 😜

**Comment by papandreou** [comment]
@camoore1979, hey, do you think you'll have time to get my PRs landed soon? 🤗 
My alternative is to create a fork and run with that, but that would make it harder to reconcile later on.

**Comment by papandreou** [comment]
Ping @camoore1979

I've published a fork with all my fixes here: https://www.npmjs.com/package/pg-to-dbml-papandreou

I have some more ideas for changes, so it'd be really nice to get some of this stuff merged.

**Review by camoore1979 — APPROVED** [review]
